encrypt/status.go

```diff
@@ -184,6 +249,9 @@ func clusterNodeStatus(res map[string]models.EncryptionStatus) (clusterStatus, e
 		}
 		if v.Mode == "IPsec" {
 			cs.EncIPsecNodeCount++
+			cs.IPsecExpectedKeyCount = v.IPsecExpectedKeyCount
```

Comment:
This silently uses the value of these fields from the last `encryptionStatus`.

Rather than store these fields on each `encryptionStatus`, they should be calculated once for the entire cluster (as mentioned above), by using the values from the first `models.EncryptionStatus` with mode "IPSec".

This eliminates some anti-patterns:

* Copying fields into each element of a collection before they are presented
* Storing overlapping data in a structure that could diverge over time

Reply:
Viktor: how about making `clusterStatus`  as a return value of `enrichWithIPsecDetails()`, where we can store `expectedKeyCount` in `clusterStatus`. In this way, we don't have to store `expectedKeyCount` in nodes' `encryptionStatus` and read from there later.

Reply:
@asauber @jschwinger233 I've refactored IPsec status but still have no idea how to get rid of `nodeStatus` struct.
Could you please review one more time?
Thanks!